### ISSUE TITLE
CDN negative queries isn't used in code

### ISSUE THREAD

**Issue by Innary** [IDEA-Research/DINO#142]
请问一下，在论文中写的
The loss to classify negative samples as background is also focal loss.
这句话，在代码中似乎CDN的负样本似乎并没有被计算损失，请问有什么原因吗，谢谢。

**Comment by HaoZhang534** [comment]
Negative examples 的box loss 没有被计算，label loss被包含在focal loss的negative部分中

**Comment by Innary** [comment]
我只在DINO.py的损失函数中看到有l_dict.update(self.get_loss(loss, output_known_lbs_bboxes, targets, dn_pos_idx, num_boxes*scalar,**kwargs))

**Comment by Innary** [comment]
我有看过这个回答，但是仍然无法理解你所说的意思。
https://github.com/IDEA-Research/DINO/issues/119#issue-1381858244
我是否可以理解成：
如果是no-object类时，就不用输入sigmoid focal loss这个损失函数中，来计算一个**targets = 全零的向量**和**outputs = sigmoid(logits_pred_class**)的sigmoid focal损失。而是直接不计算它，直接丢弃？

**Comment by HaoZhang534** [comment]
除了dn_pos_idx以外都是neg，建议debug查看这里 https://github.com/IDEA-Research/DINO/blob/main/models/dino/dino.py#L384 ，进入focal loss 之前的input和target是包含所有queries(both positive and negative)。

**Comment by Innary** [comment]
好的 谢谢~
